core/trino-main/src/test/java/io/trino/execution/scheduler/policy/TestPhasedExecutionSchedule.java

```diff
@@ -272,7 +272,7 @@ private Set<PlanFragmentId> getSchedulingFragments(PhasedExecutionSchedule sched
                 .collect(toImmutableSet());
     }
 
-    private static class TestingStageExecution
+    public static class TestingStageExecution
```

Comment:
As I mentioned above, you can have a version of this class inside `TestScaledWriterScheduler` with an implementation of `scheduleTask()` method.

Reply:
sure.